tea.go

```diff
@@ -300,9 +304,22 @@ func (p *Program) handleCommands(cmds chan Cmd) chan struct{} {
 	return ch
 }
 
+func (p *Program) recoverFromPanic() {
+	if r := recover(); r != nil {
+		fmt.Printf("Caught panic:\n\n%s\n\nRestoring terminal...\n\n", r)
+		debug.PrintStack()
+		p.errs <- fmt.Errorf("%v", r)
+		return
+	}
```

Comment:
I believe this is very close to code used in `Program.Run` so we should probably use this method there as well. That said, this method appears to be missing the `p.shutdown(true)` we see in `Program.Run`: is that intentional?

Reply:
cc: @joerdav